lib/p2p/Pool.ts

```diff
@@ -49,6 +49,7 @@ interface NodeConnectionIterator {
 class Pool extends EventEmitter {
   /** The local handshake data to be sent to newly connected peers. */
   public handshakeData!: HandshakeState;
+  public outgoingConnections: any = {};
```

Comment:
I think the best data type for this is a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) like `public outgoingConnections = new Set<string>();`. All we're interested in knowing is whether a given `nodePubKey` exists within the set.

I also feel like `pendingConnections` or maybe `pendingOutgoingConnections` would be a better name, since `outgoingConnections` to me could mean established outbound connections. Either way a brief comment explaining what this property is and what it's used for would be good.

Lastly I see this is designated as `public` - is it actually going to be used outside this class? It doesn't look like it is currently, if not it should probably be `private`.